### ISSUE TITLE
Display current time by hotkey does not show immediately or is delayed when switching off

### ISSUE THREAD

**Issue by qwertfisch** [clsid2/mpc-hc#3193]
I have used version 2.1.4 until last week, so I am sorry for not bringing this issue up sooner. I am using the current version 2.3.7 now.

I regularly use “OSD: display current time” via hotkey to have a quick check what the current time/position in the video is. So mostly I press the hotkey twice in quick succession, because I just want a quick peek via OSD. The first keypress always shows the current time on the OSD, with the second press the time disappeared immediately.

## Description

Now there are two related issues with this specific OSD information:

1. When any another information (e.g. Play/Pause status, or the video’s name) is displayed on OSD, pressing the hotkey does not have immediate effect. The application instead waits until the timeout for the other information is due (might be up to three seconds), and only then the time will be displayed.
2. When the time is displayed, pressing the hotkey to disable the option also does not have immediate effect, but needs at least one second until the OSD disappears.

## Regression

In version 2.1.4 both issues weren’t there. Activating the display of current time had an immediate effect both for enabling and for disabling this option. I also tested some releases in between:

- version 2.2.1 has already the first issue
- version 2.3.0 additionally got the second issue, probably due to changes in the OSD system

## Comments

I can guess why the second issue might be introduced: maybe the display of the time will only update once per second (which makes sense), but disabling the option should still have an immediate effect. 

The first issue seems plain wrong to me in terms of usage. Compare e.g. with another option “OSD: Show File Name”. Pressing the appropriate hotkey here _always_ immediately displays the filename, no matter what the OSD was showing before. I would expect the same behaviour also for showing the current time.

In summary this look as though the introduction of a global option "display current time" changed the OSD behaviour in an unfavorable way. I would appreciate if the previous behaviour could be reinstated.
